File: bindings/CXX11/cxx11/Attribute.cpp
/*
 * Distributed under the OSI-approved Apache License, Version 2.0.  See
 * accompanying file Copyright.txt for details.
 *
 * Attribute.cpp :
 *
 *  Created on: Jun 4, 2018
 *      Author: William F Godoy godoywf@ornl.gov
 */

#include "Attribute.h"
#include "Attribute.tcc"

namespace adios2
{

#define declare_template_instantiation(T) template class Attribute<T>;

ADIOS2_FOREACH_ATTRIBUTE_TYPE_1ARG(declare_template_instantiation)
#undef declare_template_instantiation

} // end namespace adios2

File: python/adios2/attribute.py
"""License:
  Distributed under the OSI-approved Apache License, Version 2.0.  See
  accompanying file Copyright.txt for details.
"""


class Attribute:
    """High level representation of the Attribute class in the adios2.bindings"""

    def __init__(self, io, name, *args, **kwargs):
        self.impl = io.DefineAttribute(name, *args, **kwargs)

    @property
    def impl(self):
        """Bindings implementation of the class"""
        return self._impl

    @impl.setter
    def impl(self, implementation):
        self._impl = implementation

    def __eq__(self, other):
        if isinstance(other, Attribute):
            return self.name() == other.name()
        return False

    def name(self):
        """
        Name of the Attribute

        Returns:
            Name of the Attribute as a str.
        """
        return self.impl.Name()

    def type(self):
        """
        Type of the Attribute

        Returns:
            Type of the Attribute as a str.
        """
        return self.impl.Type()

    def data(self):
        """
        Content of the Attribute

        Returns:
            Content of the Attribute as a non string.
        """
        return self.impl.Data()

    def data_string(self):
        """
        Content of the Attribute

        Returns:
            Content of the Attribute as a str.
        """
        return self.impl.DataString()
